spec-tests/src/test/java/org/candlepin/spec/content/ContentAccessSpecTest.java

```diff
@@ -0,0 +1,864 @@
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = adminClient.owners().createOwner(Owners.randomSca());
+        String ownerKey = owner.getKey();
+
+        // We expect this content to NOT be filtered out due to a match with the system's architecture
+        ContentDTO content1 = adminClient.ownerContent()
+            .createContent(ownerKey, Content.random().arches("ppc64"));
+        // We expect this content to be filtered out due to a mismatch with the system's architecture
+        ContentDTO content2 = adminClient.ownerContent()
+            .createContent(ownerKey, Content.random().arches("x86_64"));
+        // We expect this content to NOT be filtered out due it not specifying an architecture
+        ContentDTO content3 = adminClient.ownerContent()
+            .createContent(ownerKey, Content.random().arches(""));
+        ProductDTO prod = adminClient.ownerProducts().createProductByOwner(ownerKey, Products.random());
+        adminClient.ownerProducts().addContent(ownerKey, prod.getId(), content1.getId(), true);
+        adminClient.ownerProducts().addContent(ownerKey, prod.getId(), content2.getId(), true);
+        adminClient.ownerProducts().addContent(ownerKey, prod.getId(), content3.getId(), true);
+
+        adminClient.owners().createPool(ownerKey, Pools.random(prod));
+
+        ConsumerDTO consumer = adminClient.consumers().createConsumer(Consumers.random(owner)
+            .facts(Map.of("system.certificate_version", "3.3", "uname.machine", "ppc64")));
+        ApiClient consumerClient = ApiClients.ssl(consumer);
+
+        List<JsonNode> certs = consumerClient.consumers().exportCertificates(consumer.getUuid(), null);
+        assertThat(certs).singleElement();
+        Map<String, List<String>> prodIdToContentIds = CertificateUtil.toProductContentIdMap(certs.get(0));
+        assertThat(prodIdToContentIds)
+            .hasSize(1)
+            .extractingByKey("content_access", as(collection(String.class)))
+            .hasSize(2)
+            .containsExactlyInAnyOrder(content1.getId(), content3.getId());
+
+        JsonNode content = certs.get(0).get("products").get(0).get("content");
+        String certContentName = content.get(0).get("name").asText();
+        assertNotNull(certContentName);
+        JsonNode certContent1 = certContentName.equals(content1.getName()) ? content.get(0) : content.get(1);
+        JsonNode certContent3 = certContentName.equals(content3.getName()) ? content.get(0) : content.get(1);
+
+        assertEquals(content1.getType(), certContent1.get("type").asText());
+        assertEquals(content1.getName(), certContent1.get("name").asText());
+        assertEquals(content1.getLabel(), certContent1.get("label").asText());
+        assertEquals(content1.getVendor(), certContent1.get("vendor").asText());
+        assertEquals(content1.getArches(), certContent1.get("arches").get(0).asText());
```

Comment:
There are a few pieces that might be worth extracting to make tests a bit more readable. For example this might be extracted as an assertion method for comparing content/certContent.

Reply:
Good idea, I added the `compareCertContent` method to handle these assertions